samples/snippets/csharp/language-reference/tokens/string-interpolation.cs

```diff
@@ -0,0 +1,85 @@
+        CreateCultureSpecificResults();
+        Console.WriteLine();
+    }
+
+    private static void CompareWithCompositeFormatting()
+    {
+        // <Snippet1>
+        string name = "Mark";
+        var date = DateTime.Now;
+        
+        // Composite formatting:
+        Console.WriteLine("Hello, {0}! Today is {1}, it's {2:HH:mm} now.", name, date.DayOfWeek, date);
+        // String interpolation:
+        Console.WriteLine($"Hello, {name}! Today is {date.DayOfWeek}, it's {date:HH:mm} now.");
+        // Both calls produce the same output that is similar to:
+        // Hello, Mark! Today is Wednesday, it's 19:40 now.
+        // </Snippet1>
+    }
+
+    private static void AlignAndSpecifyFormat()
+    {
+        // <Snippet2>
+        Console.WriteLine($"|{"Left",-7}|{"Right",7}|");
+
+        const int FieldWidthRightAligned = 20;
+        Console.WriteLine($"{Math.PI,FieldWidthRightAligned} - default formatting of the pi number");
+        Console.WriteLine($"{Math.PI,FieldWidthRightAligned:F3} - display only three decimal digits of the pi number");
+        // Expected output is:
+        // |Left   |  Right|
+        //     3.14159265358979 - default formatting of the pi number
+        //                3.142 - display only three decimal digits of the pi number    
+        // </Snippet2>
+    }
+
+    private static void DelimitInterpolatedExpression()
+    {
+        // <Snippet3>
+        string name = "Horace";
+        int age = 34;
+        Console.WriteLine($"He asked, \"Is your name {name}?\", but didn't wait for a reply :-{{");
+        Console.WriteLine($"{name} is {age} year{(age == 1 ? "" : "s")} old.");
+        // Expected output is:
+        // He asked, "Is your name Horace?", but didn't wait for a reply :-{
+        // Horace is 34 years old.
+        // </Snippet3>
+    }
+
+    private static void CreateCultureSpecificResults()
+    {
+        // <Snippet4>
+        double price = 1000;
+        FormattableString message = $"The cost of this item is {price:C}.";
+        
+        CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("en-US");
+        string messageInCurrentCulture = message.ToString();
+
+        CultureInfo specificCulture = CultureInfo.GetCultureInfo("nl-NL");
+        string messageInSpecificCulture = message.ToString(specificCulture);
+
+        string messageInInvariantCulture = FormattableString.Invariant(message);
+
+        const int CultureOutputAlignment = -10;
+        Console.WriteLine($"{CultureInfo.CurrentCulture,CultureOutputAlignment} {messageInCurrentCulture}");
+        Console.WriteLine($"{specificCulture,CultureOutputAlignment} {messageInSpecificCulture}");
+        Console.WriteLine($"{"Invariant",CultureOutputAlignment} {messageInInvariantCulture}");
+        // Expected output is:
+        // en-US      The cost of this item is $1,000.00.
+        // nl-NL      The cost of this item is € 1.000,00.
+        // Invariant  The cost of this item is ¤1,000.00.
```

Comment:
I think this a bad example, mostly because I think currency format is a weird feature, especially when combined with current culture. An item cost won't be the same number in all currencies, so I think it doesn't make sense to use it even as an example.

One way to rewrite it would be:

```c#
double solarYearMinutes = 525948.7518;

FormattableString message = $"One solar year is {solarYearMinutes:N3} minutes.";

CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("cs-CZ");
string messageInCurrentCulture = message.ToString();

CultureInfo specificCulture = CultureInfo.GetCultureInfo("en-IN");
string messageInSpecificCulture = message.ToString(specificCulture);

string messageInInvariantCulture = FormattableString.Invariant(message);

const int CultureOutputAlignment = -10;
Console.WriteLine($"{CultureInfo.CurrentCulture,CultureOutputAlignment} {messageInCurrentCulture}");
Console.WriteLine($"{specificCulture,CultureOutputAlignment} {messageInSpecificCulture}");
Console.WriteLine($"{"Invariant",CultureOutputAlignment} {messageInInvariantCulture}");
// Expected output is:
// cs-CZ      One solar year is 525 948,752 minutes.
// en-IN      One solar year is 5,25,948.752 minutes.
// Invariant  One solar year is 525,948.752 minutes.
```

"One solar year in minutes" is weird number, but it's the best I could come up with that's sufficiently large, has fractional part and isn't currency.

The current culture is not en-US, so that current culture and invariant culture produce different output.

Reply:
Completely agree with you @svick I've just picked up another constant.
Still thinking how to update the story with dates as well.